### PR TITLE
chore(test-data): create `ProductDiscountValueRelative`, `ProductDiscountValueAbsolute`, `ProductDiscountValueExternal`, `Money`, `CentPrecisionMoney`

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

In my quest to create the `ProductDiscount` entity, we first had to create its corresponding discount `value` types.

## Description

[`ProductDiscountValueRelative`](https://docs.commercetools.com/api/projects/productDiscounts#productdiscountvaluerelative) and [`ProductDiscountValueExternal`](https://docs.commercetools.com/api/projects/productDiscounts#productdiscountvalueexternal) are self-contained and have no dependencies.

[`ProductDiscountValueAbsolute`](https://docs.commercetools.com/api/projects/productDiscounts#productdiscountvalueabsolute) required `CentPrecisionMoney`, and [`ProductDiscountValueAbsoluteDraft`](https://docs.commercetools.com/api/projects/productDiscounts#productdiscountvalueabsolutedraft) required `Money`, which were also created in this PR.

Additional comments inline.

### CODE REVIEW

File: models/cent-precision-money/src/transformers.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  }),
+  rest: Transformer<TCentPrecisionMoney, TCentPrecisionMoney>('rest', {
+    buildFields: [],
+  }),
+  graphql: Transformer<TCentPrecisionMoney, TCentPrecisionMoneyGraphql>(
+    'graphql',
+    {
+      buildFields: [],
+      addFields: () => ({
+        __typename: 'Money',
```
Comment (jaikamat) [inline models/cent-precision-money/src/transformers.ts:16]: The type is `Money` in the GraphQL API

**Review by tylermorrisford — APPROVED** [review]

File: models/money/src/builder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  );
+
+  it(
+    ...createBuilderSpec<TMoney, TMoneyGraphql>(
+      'graphql',
+      Money.random(),
+      expect.objectContaining({
+        centAmount: expect.any(Number),
+        currencyCode: expect.any(String),
+        __typename: 'BaseMoneyInput',
```
Comment (jaikamat) [inline models/money/src/builder.spec.ts:37]: The type is `BaseMoneyInput` in the GraphQL API

File: models/cent-precision-money/src/cent-precision-money-draft/transformers.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  rest: Transformer<TCentPrecisionMoneyDraft, TCentPrecisionMoneyDraft>(
+    'rest',
+    { buildFields: [] }
+  ),
+  graphql: Transformer<
+    TCentPrecisionMoneyDraft,
+    TCentPrecisionMoneyDraftGraphql
+  >('graphql', {
+    buildFields: [],
+    addFields: () => ({ __typename: 'MoneyInput' }),
```
Comment (jaikamat) [inline models/cent-precision-money/src/cent-precision-money-draft/transformers.ts:21]: Draft type deviates here from `Money` to `MoneyInput` for GraphQL

File: models/cent-precision-money/src/cent-precision-money-draft/generator.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { fake, Generator, oneOf } from '@commercetools-test-data/core';
+import { TCentPrecisionMoneyDraft } from './types';
+
+// https://docs.commercetools.com/api/types#money
```
[the remark this answers is not in the document]
Comment (jaikamat) [reply models/cent-precision-money/src/cent-precision-money-draft/generator.ts]: Thanks! 2173aec1e0c8cfc488ff8f6420c0398f8894cfba

**Comment by jaikamat** [comment]
@valoriecarli see 0b589d61dd3c6ff9f848a6ae90a9b177779f0649

File: models/cent-precision-money/src/generator.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { fake, Generator, oneOf } from '@commercetools-test-data/core';
+import { TCentPrecisionMoney } from './types';
+
+// https://docs.commercetools.com/api/types#centprecisionmoney
+
+const generator = Generator<TCentPrecisionMoney>({
+  fields: {
+    type: 'centPrecision',
+    currencyCode: oneOf('EUR', 'USD'),
```
[the remark this answers is not in the document]
Comment (jaikamat) [reply models/cent-precision-money/src/generator.ts]: Had a slack conversation about this. I don't have opinions...yet. Agreed to keep this in the back of our minds and iterate going forward 😄

File: models/cent-precision-money/src/cent-precision-money-draft/builder.spec.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/cent-precision-money/src/cent-precision-money-draft/builder.spec.ts:2]: Are these needed?

File: models/money/package.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "@commercetools-test-data/money",
+  "version": "4.4.0",
+  "description": "Data model for commercetools API CentPrecisionMoney",
```
Comment (stephsprinkle) [inline models/money/package.json:4]:
```suggestion
  "description": "Data model for commercetools API Money",
```
???

File: models/money/src/builder.spec.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/money/src/builder.spec.ts:2]: Same question

File: models/product-discount-value-absolute/src/builder.spec.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/product-discount-value-absolute/src/builder.spec.ts:2]: Again!

File: models/product-discount-value-absolute/src/product-discount-value-absolute-draft/builder.spec.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/product-discount-value-absolute/src/product-discount-value-absolute-draft/builder.spec.ts:2]: 😉

**Review by tylermorrisford — APPROVED** [review]

File: models/product-discount-value-external/src/builder.spec.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/product-discount-value-external/src/builder.spec.ts:2]: Bee-boop.

File: models/product-discount-value-external/src/product-discount-value-external-draft/builder.spec.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable jest/no-disabled-tests */
```
Comment (stephsprinkle) [inline models/product-discount-value-external/src/product-discount-value-external-draft/builder.spec.ts:1]: Bloop

File: models/product-discount-value-relative/src/builder.spec.ts [hunk 13]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/product-discount-value-relative/src/builder.spec.ts:2]: 🤖

File: models/product-discount-value-relative/src/product-discount-value-relative-draft/builder.spec.ts [hunk 14]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable jest/valid-title */
```
Comment (stephsprinkle) [inline models/product-discount-value-relative/src/product-discount-value-relative-draft/builder.spec.ts:2]: 🦾

**Review by stephsprinkle — APPROVED** [review]
Seems reasonable, some minor stuff 😉

File: models/money/package.json (hunk 7 above)
Comment (jaikamat) [reply models/money/package.json:4]: Rogue copypasta strikes again!

File: models/money/src/builder.spec.ts (hunk 8 above)
Comment (jaikamat) [reply models/money/src/builder.spec.ts:2]: Addressed in 588274198e7103b275307fe755d79761bad755e3

File: models/money/package.json (hunk 7 above)
Comment (jaikamat) [reply models/money/package.json]: Addressed in 588274198e7103b275307fe755d79761bad755e3

File: models/product-discount-value-relative/src/product-discount-value-relative-draft/builder.spec.ts (hunk 14 above)
Comment (jaikamat) [reply models/product-discount-value-relative/src/product-discount-value-relative-draft/builder.spec.ts:2]: We do need these, my linter complains if they're omitted

**Review by tdeekens — APPROVED** [review]

**Pull request merged**